### PR TITLE
i2c: cpp check fixes

### PR DESCRIPTION
**Pull request by dylad**
### Contribution description
This PR should fix all CPP issues for new_i2c_if feature branch, I cherry-picked some commits from @smlng and #9571. I also include the required changes from @aabadie (see #9571 discussion)

Once this PR get merged, I'll rebase new_i2c_if3 to see if Murdock is happy.
This way it should ease comparison between original new_i2c_if branch and new_i2c_if3 which will be rebased onto master.

### Issues/PRs references
#9571

### CODE REVIEW

**Comment by dylad** [comment]
Murdock still fails because ublox and EFM32 boards need to be rebased onto master but I prefer rebase a cloned branch rather than this branch to keep tracks of the changes.

**Comment by aabadie** [comment]
Thanks @dylad, regarding STM32F1, this is ok now.

File: boards/acd52832/include/periph_conf.h [hunk 1]
```diff
@@ -104,7 +104,8 @@ static const i2c_conf_t i2c_config[] = {
     {
         .dev = NRF_TWIM1,
         .scl = 28,
-        .sda = 29
+        .sda = 29,
+        .speed = I2C_SPEED_NORMAL
```
Comment (cladmi) [inline boards/acd52832/include/periph_conf.h]: As you said, this one is actually more than a warning as I2C_SPEED_NORMAL is actually not 0. Thanks cpp check :)

**Comment by cladmi** [comment]
> Murdock still fails because ublox and EFM32 boards need to be rebased onto master but I prefer rebase a cloned branch rather than this branch to keep tracks of the changes.

You mean that it is just Murdock that fails, and would fail without it anyway and that it can be merged even with the error ?

The changes look good to me.

**Comment by dylad** [comment]
This PR fixes a lot of issues with Murdock (all CPPcheck in fact) but with or without this PR, Murdock fails with EFM32 based board and the ublox board. Theses failures are not related to this PR. 
And the remaining Murdock failures will be fix with the rebase.

**Comment by cladmi** [comment]
Ok that what I was thinking but was not sure.

**Pull request merged**